common/vm_wait_network_connected.yml

```diff
@@ -32,10 +18,43 @@
     name: "{{ vm_name }}"
     gather_network_info: true
   register: vm_netadapter_facts
-  until: "'network_data' in vm_netadapter_facts and vm_netadapter_facts.network_data[item].connected"
+  until:
+    - vm_netadapter_facts is defined
+    - vm_netadapter_facts.network_info is defined
+    - vm_netadapter_facts.network_info | selectattr('start_connected', 'equalto', true) | rejectattr('connected', 'equalto', true) | length == 0
   retries: "{{ ((vm_wait_network_connect_timeout | int) / 3) | int }}"
   delay: 3
-  with_items: "{{ start_connected_net_adapter }}"
+  ignore_errors: true
+
 - name: "Print the network adapter info"
   ansible.builtin.debug: var=vm_netadapter_facts
   when: enable_debug is defined and enable_debug
+
+- name: "VM has start connected VM network adapters disconnected"
+  block:
+    - name: "Initialize the list of start connected network adapters failed to be connected"
+      ansible.builtin.set_fact:
+        fail_connected_netadapter_info: []
+
+    - name: "Update the list of start connected network adapters failed to be connected"
+      ansible.builtin.set_fact:
+        fail_connected_netadapter_info: >-
+          {{
+            vm_netadapter_facts.network_info |
+            selectattr('start_connected', 'equalto',true) |
+            rejectattr('connected', 'equalto', true) |
+            sort(attribute='label')
+          }}
+      when:
+        - vm_netadapter_facts is defined
+        - vm_netadapter_facts.network_info is defined
+
+    - name: "Waiting for network adapter connected failed"
+      ansible.builtin.fail:
+        msg: >-
```

Comment:
If `vm_netadapter_facts.failed` just true, and no `vm_netadapter_facts.network_info` returned, this `        fail_connected_netadapter_info: []` is empty list, then this fail_msg will fail to print

Reply:
I tested it, which didn't fail.
```
- name: play1
  hosts: localhost
  vars:
    fail_connected_netadapter_info: []
  tasks:
    - debug:
        msg: >-
          It's timed out to wait for start connected network adapters
          '{{ ",".join(fail_connected_netadapter_info | map(attribute="label")) }}' to be connected

```
```
PLAY [play1] **********************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************
ok: [localhost]

TASK [debug] **********************************************************************************************************
ok: [localhost] => {
    "msg": "It's timed out to wait for start connected network adapters '' to be connected"
}
```